policy-engine/src/graph.rs

```diff
@@ -40,59 +39,65 @@ lazy_static! {
 }
 
 /// Serve Cincinnati graph requests.
-pub(crate) fn index(req: HttpRequest<AppState>) -> Box<Future<Item = HttpResponse, Error = Error>> {
+pub(crate) fn index(
+    req: HttpRequest<AppState>,
+) -> Box<Future<Item = HttpResponse, Error = GraphError>> {
     HTTP_GRAPH_REQS.inc();
 
+    // Check that the client can accept JSON media type.
+    if let Err(e) = commons::ensure_content_type(req.headers(), CONTENT_TYPE) {
+        HTTP_GRAPH_BAD_REQS.inc();
+        return Box::new(future::err(e));
+    }
+
     // Check for required client parameters.
     let mandatory_params = &req.state().mandatory_params;
-    if commons::ensure_query_params(mandatory_params, req.query_string()).is_err() {
+    if let Err(e) = commons::ensure_query_params(mandatory_params, req.query_string()) {
```

Comment:
opportunity: combine this into something declarative with the previous error check